src/platform/Darwin/system/SystemLayerImplDispatch.mm

```diff
@@ -258,7 +266,17 @@ static void TimerCompleteBlockCallback(Layer * aLayer, void * appState)
 #if SYSTEM_LAYER_IMPL_DISPATCH_DEBUG
         ChipLogError(Inet, "%s (onComplete: %p - appState: %p)", __func__, onComplete, appState);
 #endif
+        dispatch_time_t walltime = DISPATCH_WALLTIME_NOW;
+        const TimerData * data = FindTimerData(onComplete, appState);
```

Comment:
Is there a specific benefit to having a list of dispatch_source_t as the primary data structure and storing the extra info we need as the context, vs just holding a list of TimerData directly and having that contain a reference to the dispatch_source_t?